internal/cmds/cmds.go

```diff
@@ -334,21 +336,30 @@ func checkAndSaveSeqNum(ctx log.Logger, seq int, mrseqPath string) (shouldExit b
 // Copy state of the extension from old version to new version during update (.mrseq files, .status files)
 func CopyStateForUpdate(ctx log.Logger) error {
 	// Copy .mrseq files (Most Recently executed Sequence number) that helps determine whether a sequence number of Run Command has been previously executed or not.
-	err := copyFiles(ctx, ".mrseq", "")
-	if err != nil {
-		return err
+	mrseqFilesNameList, mrseqFileCopyErr := copyFiles(ctx, ".mrseq", "")
+	if mrseqFileCopyErr != nil {
+		return mrseqFileCopyErr
 	}
 
 	// Copy .status files of already executed sequence numbers
-	err = copyFiles(ctx, ".status", "status")
-	if err != nil {
-		return err
+	_, statusFileCopyErr := copyFiles(ctx, ".status", constants.StatusFileDirectory)
+	if statusFileCopyErr != nil {
+		return statusFileCopyErr
 	}
+
+	// If status file corresponding to a .mrseq file does not exist, create a dummy status file to prevent poll status timeouts for already executed Run Commands after upgrade.
+	if mrseqFilesNameList != nil && mrseqFilesNameList.Len() > 0 {
+		createDummyStatusFilesErr := createDummyStatusFilesIfNeeded(ctx, mrseqFilesNameList)
+		if createDummyStatusFilesErr != nil {
```

Comment:
I'm wondering if this should be best effort. What if there's a race condition where an executing RC tries to write the file? Best effort (perhaps with a trace) seems like a safe approach.

Reply:
Yes, I agree it should be best effort. Changed accordingly. Added traces inside createDummyStatusFilesNeeded()
Race condition - I do not expect a race condition to happen. createDummyStatusFilesNeeded() is called only during Update operation. Enable operations that can write to status file only comes later once Update has completed.